File: python/credit_card_validator/credit_card_validator.py


def card_length(card_number):

    card_digit = []
    
    for digit in str(card_number):
        card_digit.append(int(digit))    
    
    return len(card_digit)
        
   

def card_type(card_number):

    card_digit = []
    
    for digit in str(card_number):
        card_digit.append(int(digit)) 
        
    if(card_digit[0] == 4):
        return "Visa Card"
       
    elif(card_digit[0] == 5):
        return "Master Card"
       
    elif(card_digit[0] == 6):
        return "Discover card"
       
    else:
        return "No type Discover"
        


def validating_card(card_number):

    card_digit = []
    
    for digit in str(card_number):
        card_digit.append(int(digit))         

    odd_index_sum = 0;

    for digit in range(1, len(card_digit), 2):

        odd_index_sum += card_digit[digit] 
        
            

    even_index_sum = 0;

    for digit in range(0, len(card_digit), 2):

        even_index =  2 * (card_digit[digit])

        if even_index > 9:
        
            for number in str(even_index):
                number = int(number)    
                even_index_sum = even_index_sum + (number % 10)  

        else:
        
            even_index_sum = even_index_sum + even_index
 

    even_and_odd_sum = (even_index_sum + odd_index_sum);
    
    if even_and_odd_sum % 10 == 0:
            
        return "Valid card"
    else:

        return "Invalid card"

File: java/CreditCardValidator/CreditCardValidator.java

import java.util.ArrayList;

public class CreditCardValidator{


    public static int cardLength(long cardNumber){
    
        ArrayList<Long> cardDigit = new ArrayList<>();
        
        for(long count = cardNumber; 0 < count; count/=10){
        
            cardDigit.add(count % 10);
            
        }
            return cardDigit.size();
    }    



    public static String cardType(long cardNumber){
    
        ArrayList<Long> cardDigit = new ArrayList<>();
        
        for(long count = cardNumber; 0 < count; count/=10){
        
            cardDigit.add(count % 10);    
            
            } 
        
            if(cardDigit.get(cardDigit.size()-1) == 4){
               return "Visa Card";}
               
            else if(cardDigit.get(cardDigit.size()-1) == 5){
               return "Master Card"; }
               
            else if(cardDigit.get(cardDigit.size()-1) == 6)
               return "Discover card";
               
             else{
                  return "No type Discover";
             }  
    }    




    public static String validatingCard(long cardNumber){
    
        ArrayList<Long> cardDigit = new ArrayList<>();
        
        for(long count = cardNumber; 0 < count; count/=10){
        
            cardDigit.add(count % 10);
        
        }
        
            int evenIndexSum = 0;
         for(int index = 0; index < cardDigit.size(); index +=2){
             
            evenIndexSum += cardDigit.get(index); 
         
         } 
        
            long oddIndexSum = 0;
         for(int index = 1; index < cardDigit.size(); index +=2){
            
               long oddIndex =  (2 * cardDigit.get(index));

                if(oddIndex > 9)
                    for(long count = oddIndex; 0 < count; count /= 10){
                    
                    oddIndexSum = oddIndexSum + (count % 10);    
                    
                    }
                    
                else{
                
                    oddIndexSum = oddIndexSum + oddIndex; 
         
                    }
            } 
        
            long evenAndOddSum = (evenIndexSum + oddIndexSum);
            if(evenAndOddSum % 10 == 0)
                return "Valid card";    
            else{
                return "Invalid card";
            }

    }   
    

}    
